pkgs/trebuchet/bin/trebuchet.dart

```diff
@@ -0,0 +1,212 @@
+// Copyright (c) 2024, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:io';
+
+import 'package:args/args.dart';
+import 'package:path/path.dart' as p;
+
+Future<void> main(List<String> arguments) async {
+  final argParser = ArgParser()
+    ..addOption(
+      'input-name',
+      help: 'Name of the package which should be transferred to a mono-repo',
+    )
+    ..addOption(
+      'input-path',
+      help: 'Path to the package which should be transferred to a mono-repo',
+    )
+    ..addOption(
+      'target-path',
+      help: 'Path to the mono-repo',
+    )
+    ..addOption(
+      'branch-name',
+      help: 'The name of the main branch on the input repo',
+      defaultsTo: 'main',
+    )
+    ..addOption(
+      'git-filter-repo',
+      help: 'Path to the git-filter-repo tool',
+    )
+    ..addFlag(
+      'push',
+      help: 'Whether to push the branch to remote',
+      defaultsTo: true,
+    )
+    ..addFlag(
+      'help',
+      abbr: 'h',
+      help: 'Prints usage info',
+      negatable: false,
+    )
+    ..addFlag(
+      'dry-run',
+      help: 'Do not actually execute any of the steps',
+      defaultsTo: false,
+    );
+
+  String input;
+  String inputPath;
+  String targetPath;
+  String branchName;
+  String gitFilterRepo;
+  bool push;
+  bool dryRun;
+  try {
+    final parsed = argParser.parse(arguments);
+    if (parsed.flag('help')) {
+      print(argParser.usage);
+      exit(0);
+    }
+
+    input = parsed['input-name'] as String;
```

Comment:
You can now use `parsed.option('input-name)` instead of grabbing the option out of a map and casting; it's slightly cleaner.